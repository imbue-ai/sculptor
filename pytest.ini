[pytest]
asyncio_mode = auto
filterwarnings =
    ignore::DeprecationWarning:wandb.*:
    ignore::DeprecationWarning:graphql.*:
    ignore::DeprecationWarning:notebook.*:
    ignore::DeprecationWarning:nptyping.*:
markers =
    # we could consider using @pytest.mark.timeout markers instead of these, if there's a way to filter by them.
    # unmarked tests should be <1 second
    up_to_10_seconds: should complete in 10 seconds
    up_to_30_seconds: should complete in 30 seconds
    up_to_5_minutes: should complete in 5 minutes
python_classes =
addopts =
    --ignore=quarantine
    --ignore-glob=**/test_data/**
    --snapshot-ignore-file-extensions=llm_cache_json
norecursedirs = quarantine
