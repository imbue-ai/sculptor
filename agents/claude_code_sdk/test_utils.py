import json
from typing import Any

import pytest
from syrupy.assertion import SnapshotAssertion

from imbue_core.sculptor.state.claude_state import parse_imbue_cli_content


@pytest.mark.parametrize(
    "tool_result",
    [
        {
            "type": "text",
            "text": '[\n  {\n    "command": "verify",\n    "issues": [\n      {\n        "key": {\n          "command": "imbue_verify",\n          "identifier": "correctness_syntax_issues:sculptor/sculptor/agents/claude_code_sdk/utils.py: 167-170"\n        },\n        "severity": "CRITICAL"\n      },\n      {\n        "key": {\n          "command": "imbue_verify",\n          "identifier": "incomplete_integration_with_existing_code:sculptor/sculptor/agents/claude_code_sdk/utils.py: 161"\n        },\n        "severity": "CRITICAL"\n      },\n      {\n        "key": {\n          "command": "imbue_verify",\n          "identifier": "documentation_implementation_mismatch:sculptor/sculptor/agents/claude_code_sdk/utils.py: 162"\n        },\n        "severity": "NIT"\n      },\n      {\n        "key": {\n          "command": "imbue_verify",\n          "identifier": "logic_error:sculptor/sculptor/agents/claude_code_sdk/utils.py: 165-170"\n        },\n        "severity": "CRITICAL"\n      },\n      {\n        "key": {\n          "command": "imbue_verify",\n          "identifier": "runtime_error_risk:sculptor/sculptor/agents/claude_code_sdk/utils.py: 170"\n        },\n        "severity": "CRITICAL"\n      },\n      {\n        "key": {\n          "command": "imbue_verify",\n          "identifier": "logic_error:sculptor/sculptor/agents/claude_code_sdk/utils.py: 161-172"\n        },\n        "severity": "ERROR"\n      }\n    ],\n    "summary": null\n  }\n]',
        },
        {
            "type": "text",
            "text": "[]",
        },
        {"type": "text", "text": '[{"command": "verify"}]'},
    ],
)
def test_parse_imbue_cli_content(tool_result: dict[str, Any], snapshot: SnapshotAssertion) -> None:
    issues, summary = parse_imbue_cli_content(tool_result)

    assert summary == snapshot
    assert issues == snapshot


def retrieve_tool_result(retrieve_result: str) -> dict[str, Any]:
    tool_result = {
        "type": "text",
        "text": json.dumps(
            [
                {
                    "command": "retrieve",
                    "summary": retrieve_result,
                    "issues": [],
                    "user_display": {"objectType": "CommandTextOutput", "output": retrieve_result},
                }
            ]
        ),
    }
    return tool_result


def test_parse_imbue_cli_content_retrieve(snapshot: SnapshotAssertion) -> None:
    tool_name = "mcp__imbue_tools__retrieve"

    # test multiple results, one result, and no results
    issues_0, summary_0 = parse_imbue_cli_content(retrieve_tool_result("test.py\ntest.js"), tool_name)
    issues_1, summary_1 = parse_imbue_cli_content(retrieve_tool_result("test.py"), tool_name)
    issues_2, summary_2 = parse_imbue_cli_content(retrieve_tool_result(""), tool_name)

    assert summary_0 == snapshot
    assert issues_0 == snapshot
    assert summary_1 == snapshot
    assert issues_1 == snapshot
    assert summary_2 == snapshot
    assert issues_2 == snapshot
