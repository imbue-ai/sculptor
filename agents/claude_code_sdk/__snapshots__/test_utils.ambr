# serializer version: 1
# name: test_parse_imbue_cli_content[tool_result0]
  "Check 'verify' found 6 issues"
# ---
# name: test_parse_imbue_cli_content[tool_result0].1
  ImbueCLIToolContent(content_type='imbue_cli', action_outputs=[ImbueCLIActionToolContent(content_type='imbue_cli_action', check='verify', user_display=None, summary=None, issues=[ImbueCLIActionIssue(description='', severity='CRITICAL', location=None, object_type=None, debug_full_details='{"key": {"command": "imbue_verify", "identifier": "correctness_syntax_issues:sculptor/sculptor/agents/claude_code_sdk/utils.py: 167-170"}, "severity": "CRITICAL"}'), ImbueCLIActionIssue(description='', severity='CRITICAL', location=None, object_type=None, debug_full_details='{"key": {"command": "imbue_verify", "identifier": "incomplete_integration_with_existing_code:sculptor/sculptor/agents/claude_code_sdk/utils.py: 161"}, "severity": "CRITICAL"}'), ImbueCLIActionIssue(description='', severity='NIT', location=None, object_type=None, debug_full_details='{"key": {"command": "imbue_verify", "identifier": "documentation_implementation_mismatch:sculptor/sculptor/agents/claude_code_sdk/utils.py: 162"}, "severity": "NIT"}'), ImbueCLIActionIssue(description='', severity='CRITICAL', location=None, object_type=None, debug_full_details='{"key": {"command": "imbue_verify", "identifier": "logic_error:sculptor/sculptor/agents/claude_code_sdk/utils.py: 165-170"}, "severity": "CRITICAL"}'), ImbueCLIActionIssue(description='', severity='CRITICAL', location=None, object_type=None, debug_full_details='{"key": {"command": "imbue_verify", "identifier": "runtime_error_risk:sculptor/sculptor/agents/claude_code_sdk/utils.py: 170"}, "severity": "CRITICAL"}'), ImbueCLIActionIssue(description='', severity='ERROR', location=None, object_type=None, debug_full_details='{"key": {"command": "imbue_verify", "identifier": "logic_error:sculptor/sculptor/agents/claude_code_sdk/utils.py: 161-172"}, "severity": "ERROR"}')])])
# ---
# name: test_parse_imbue_cli_content[tool_result1]
  'Executed 0 checks and found no issues'
# ---
# name: test_parse_imbue_cli_content[tool_result1].1
  ImbueCLIToolContent(content_type='imbue_cli', action_outputs=[])
# ---
# name: test_parse_imbue_cli_content[tool_result2]
  "Check 'verify' found no issues"
# ---
# name: test_parse_imbue_cli_content[tool_result2].1
  ImbueCLIToolContent(content_type='imbue_cli', action_outputs=[ImbueCLIActionToolContent(content_type='imbue_cli_action', check='verify', user_display=None, summary=None, issues=[])])
# ---
# name: test_parse_imbue_cli_content_retrieve
  'test.py...'
# ---
# name: test_parse_imbue_cli_content_retrieve.1
  ImbueCLIToolContent(content_type='imbue_cli', action_outputs=[ImbueCLIActionToolContent(content_type='imbue_cli_action', check='retrieve', user_display=None, summary='test.py\ntest.js', issues=[])])
# ---
# name: test_parse_imbue_cli_content_retrieve.2
  'test.py'
# ---
# name: test_parse_imbue_cli_content_retrieve.3
  ImbueCLIToolContent(content_type='imbue_cli', action_outputs=[ImbueCLIActionToolContent(content_type='imbue_cli_action', check='retrieve', user_display=None, summary='test.py', issues=[])])
# ---
# name: test_parse_imbue_cli_content_retrieve.4
  'No files found matching the query'
# ---
# name: test_parse_imbue_cli_content_retrieve.5
  ImbueCLIToolContent(content_type='imbue_cli', action_outputs=[ImbueCLIActionToolContent(content_type='imbue_cli_action', check='retrieve', user_display=None, summary='', issues=[])])
# ---
